### ISSUE TITLE
mAP on training with multiple datasets --- Discussion 

### ISSUE THREAD

**Issue by ghost** [jixunbo/LightMBN#5]
Thank you for being patient with my bombarding questions ;) I'm wondering if you can point me in a direction that can show what I am missing in this process.

I have actually trained the model with market1501+dukemtmc+sensereid+cuhk03+grid+custom datasets and actually achieved a mAP of 0.3304 after training for 130 epochs --- mAP was still improving although very slowly .

1) Noticed that the number of classes does not change : it is always 751 irrespective of dataset -- the readme says it is automatically adjusted with the data. Will this be reflected in the config file ?

2) Maybe this is a problem of plenty but which loss functions would you recommend ? I'm usually sticking with the example given in readme ---  0.5*CrossEntropy+0.5*MSLoss

Any suggestions as to how you went about getting the best performance in terms of mAP. 

Much appreciation to your repository again. Hopefully my questions will help others using the repo too. 

Thank You

**Comment by jixunbo** [comment]
hi,
for the 1. question, it would not be reflected in the config file, but in the printed dataset summary you could find the 'train ids' is automatically changed,
for the 2. question, as you can see in our paper we compared CE+MS and CE+Tri, but it would be also interesting if you try other combinations.
I have not trained on such multiple datasets but I could imagine the performance would be much lower since the datasets are not in the same domain, there exists great bias(eg. lighting condition viewing angles different number of images etc.) between them, it is another research problem that how to synthesize datasets in the proper way. You may need to read some more papers regarding domain transfer/multi-domain learning.:slightly_smiling_face:

**Comment by gayatriprasad** [comment]
Thank you for the reply. 

I have actually trained on mlutiple datasets. One thing I observe is the model although it has not given a high mAP for test, it does seem to have generalized well. It gives better performance when tested on new data.
